Enable aten.upsample_nearest2d.vec

cccclai opened the pull request:
Summary:
As title, fix some edge cases and add two unit tests
```
buck test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_up_sampling_nearest_2d_with_size (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNQuantizedOperatorSimulator)'
buck test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_up_sampling_nearest_2d_with_size (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNFloatingPointOperatorSimulator)'
```

Differential Revision: D70796006

haowhsu-quic commented on backends/qualcomm/tests/test_qnn_delegate.py:
```diff
@@ -1430,6 +1440,16 @@ def test_qnn_backend_interpolate_nearest_2d(self):
+    def test_qnn_backend_up_sampling_nearest_2d_with_scale_factor(self):
+        module = UpsampleNearest2D(scale_factor=2)
+        sample_input = (torch.randn(1, 16, 72, 104), )
+        self.lower_module_and_test_output(module, sample_input)
```
I guess you might need this for quantized operator:
```
module = self.get_qdq_module(module, sample_input)
```

cccclai replied:
thanks - updated

cccclai committed:
Enable aten.upsample_nearest2d.vec (#9067)

Summary:

As title, fix some edge cases and add two unit tests
```
buck test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_up_sampling_nearest_2d_with_size (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNQuantizedOperatorSimulator)'
buck test 'fbcode//mode/dev-nosan' fbcode//executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator -- --exact 'executorch/backends/qualcomm/tests/fb:test_qnn_delegate_simulator - test_qnn_backend_up_sampling_nearest_2d_with_size (executorch.backends.qualcomm.tests.fb.test_qnn_delegate_simulator.TestQNNFloatingPointOperatorSimulator)'
```

Differential Revision: D70796006
